### ISSUE TITLE
Bert model returns {batchsize, sequence_length, vocabulary_size} dimensional latent_report & not {batchsize, sequence_length}.

### ISSUE THREAD

**Issue by one2clouds** [SZUHvern/MaCo#3]
Hello, thankyou for making this valuable project open source. 

I have a problem with your code. 
In model_MaCo.py, line number 399 & 400,
the output we get from BERT model is of dimension {batchsize, sequence_length, vocabulary_size} 
& currently you have mentioned it is of size {batchsize, sequence_length}
How did you remove the 3rd dimension?

**Comment by one2clouds** [comment]
```
From line 394 : 

        # MaCo
        latent_img = self.img_mlp(latent)
        latent_img = latent_img[:, 1:, :] # Removing the CLS token, at dim 1
        latent_img_global = latent_img.mean(dim=1)  # mean across patches to have global patch embedding 
        labels = None
        latent_report = self.bert_encoder(latent_img_global, ids, labels, attention_mask, type_ids)
        logits_report = latent_report.logits
        x_ = self.pos_weight_img(x_pos).type(torch.float32)

        latent_img_global = gather_gradient(latent_img_global)
        logits_report = gather_gradient(logits_report)
        x_ = gather_gradient(x_)

        # print(logits_report.shape) # torch.Size([64, 100, 30000]) # {batch_size, sequence_len, num_tokens} 


        ##<<< ------------------------!!!!!!!! Updated Code Start!!!!!!! -------------------------------------------------->>>>>_##

        predicted_tokenes = torch.argmax(logits_report, dim=-1)

        ##<<< ------------------------ !!!!!! Updated Code Stop !!!!!!!!!-------------------------------------------------->>>>>_##



        logits1, logits2 = create_logits(latent_img_global, predicted_tokenes, self.Tpar)
        gt = torch.arange(logits1.shape[0], dtype=torch.long).cuda()
```

Does doing argmax make sense, in our approach? After calculating LOSS?

**Comment by one2clouds** [comment]
Sorry, but inside bert/bert.py. In line 287 & 201 had different code than yours, which caused error. I have fixed the error now.....
```
        sequence_poolput = outputs[1]
        return MaskedLMOutput(
            loss=masked_lm_loss,
            logits=sequence_poolput,
            hidden_states=outputs.hidden_states,
            attentions=outputs.attentions,
        )
```
